npm_and_yarn/lib/dependabot/npm_and_yarn.rb

```diff
@@ -25,3 +25,157 @@
     groups.include?("dependencies")
   end
 )
+
+module Dependabot
+  module NpmAndYarn
+    YARN_CODE_REGEX = /(YN\d{4})/
+    YARN_ERROR_CODES = T.let({
+      "YN0001" => {
+        message: "Exception error",
+        new_error: ->(_error, message) { Dependabot::DependabotError.new(message) }
+      },
+      "YN0002" => {
+        message: "Missing peer dependency",
+        new_error: ->(_error, message) { Dependabot::DependencyFileNotResolvable.new(message) }
+      },
+      "YN0016" => {
+        message: "Remote not found",
+        new_error: ->(_error, message) { Dependabot::GitDependenciesNotReachable.new(message) }
+      },
+      "YN0020" => {
+        message: "Missing lockfile entry",
+        new_error: ->(_error, message) { Dependabot::DependencyFileNotFound.new(message) }
+      },
+      "YN0046" => {
+        message: "Automerge failed to parse",
+        new_error: ->(_error, message) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
+      },
+      "YN0047" => {
+        message: "Automerge immutable",
+        new_error: ->(_error, message) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
+      },
+      "YN0062" => {
+        message: "Incompatible OS",
+        new_error: ->(_error, message) { Dependabot::IncompatibleOS.new(message) }
```

Comment:
I just checked the incompatible OS error and it seems there is little we can do here; should we make this a Dependabot error? Seems we are trying to run the user's package json on our host but we can't do that cos of mismatching arch?

Reply:
Removed `Incompatible OS`